nav2_system_tests/src/updown/test_updown.cpp

```diff
@@ -0,0 +1,107 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <random>
+#include <vector>
+
+#include "geometry_msgs/msg/pose_stamped.hpp"
+#include "nav2_lifecycle_manager/lifecycle_manager_client.hpp"
+#include "rcutils/cmdline_parser.h"
+
+using namespace std::chrono_literals;
+
+struct xytheta
+{
+  double x;
+  double y;
+  double theta;
+};
+
+int main(int argc, char ** argv)
+{
+  rclcpp::init(argc, argv);
+  nav2_lifecycle_manager::LifecycleManagerClient client;
+
+  // Create a set of target poses across the map
+  std::vector<xytheta> target_poses;
+  target_poses.push_back({-2.0, -0.5, 0});
+  target_poses.push_back({0.94, -0.55, 0});
+  target_poses.push_back({1.7, 0.5, 1.4});
+  target_poses.push_back({0.97, 1.68, 2.94});
+  target_poses.push_back({0.02, 1.74, -2.9});
+
+  xytheta & initial_pose = target_poses[0];
+
+  // Wait for a few seconds to let all of the nodes come up
+  std::this_thread::sleep_for(5s);
+
+  // Start the nav2 system, bringing it to the ACTIVE state
+  client.startup();
+
+  // Set the robot's starting pose (approximately where it comes up in gazebo)
+  client.set_initial_pose(initial_pose.x, initial_pose.y, initial_pose.theta);
+
+  // Wait for a couple secs to let the rviz display update
```

Comment:
If this is meant to be a automated test, Rviz shouldn't be running, is this sleep_for(2s) still needed?

Reply:
```suggestion
  // Wait for a couple secs to let the pose update
```

Reply:
I originally had this in there to make sure that the node discovery traffic had settled out before starting the test. I'd like to keep it for now and then remove it when the test is running solidly. Recently, it hasn't been 100% with the recent ROS changes.